### PR TITLE
Switch ugly password boxes to Field or styled input

### PR DESCRIPTION
**Pull request by t3chguy**
image
image
image
image

Partially fixes https://github.com/vector-im/riot-web/issues/4929

Usages for password fields of `<input` vs `<Field` is inconsistent throughout codebase, here it just felt nicer that on such a barren modal a Field is used IMO. If reviewer prefers I switch all over to Field then sure. Also removes a usage of a ref because they're discouraged in react in favour of controlled components.

Signed-off-by: Michael Telatynski <7t3chguy@gmail.com>

### CODE REVIEW

File: res/css/views/dialogs/_DeactivateAccountDialog.scss [hunk 1]
```diff
@@ -21,3 +21,10 @@ limitations under the License.
 .mx_DeactivateAccountDialog .mx_DeactivateAccountDialog_input_section {
     margin-top: 60px;
 }
+
+.mx_DeactivateAccountDialog .mx_DeactivateAccountDialog_input_section input {
```
Comment (jryans) [inline res/css/views/dialogs/_DeactivateAccountDialog.scss]: I think I would indeed like to see this one become a `Field` as well. Is there some reason you didn't want to do that for this one?

File: res/css/views/auth/_InteractiveAuthEntryComponents.scss [hunk 2]
```diff
@@ -55,4 +55,12 @@ limitations under the License.
...
-}
\ No newline at end of file
+}
+
+.mx_InteractiveAuthEntryComponents_passwordSection {
+    width: 300px;
+}
+
+.mx_InteractiveAuthEntryComponents_passwordSection input {
+    width: 100%;
```
Comment (jryans) [inline res/css/views/auth/_InteractiveAuthEntryComponents.scss]: If you're interested, we could probably remove most of these `width: X` styles for all of the inputs inside `Field`s by doing https://github.com/vector-im/riot-web/issues/9344.

File: src/components/views/auth/InteractiveAuthEntryComponents.js [hunk 3]
```diff
@@ -138,17 +130,20 @@ export const PasswordAuthEntry = React.createClass({
...
         return (
             <div>
                 <p>{ _t("To continue, please enter your password.") }</p>
-                <form onSubmit={this._onSubmit}>
-                    <label htmlFor="passwordField">{ _t("Password:") }</label>
-                    <input
-                        name="passwordField"
-                        ref="passwordField"
+                <form onSubmit={this._onSubmit} className="mx_InteractiveAuthEntryComponents_passwordSection">
+                    <Field
```
Comment (jryans) [inline src/components/views/auth/InteractiveAuthEntryComponents.js]: Please also add an `id` prop to bind the label and input together, such as:

```
id="mx_InteractiveAuthEntryComponents_password"
```

File: res/css/views/dialogs/_DeactivateAccountDialog.scss (hunk 1 above)
Comment (t3chguy) [reply res/css/views/dialogs/_DeactivateAccountDialog.scss]: I tried to do as few changes but the label on the other field just irked me so much, I'll switch them both over, FIeld > input :D

Comment (t3chguy) [reply res/css/views/dialogs/_DeactivateAccountDialog.scss]: image

**Pull request merged**